gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-core/src/main/java/io/gravitee/am/gateway/handler/root/service/user/impl/UserServiceImpl.java

```diff
@@ -286,7 +297,11 @@ public Single<ResetPasswordResponse> resetPassword(Client client, User user, io.
                     // see https://github.com/gravitee-io/issues/issues/8407
                     return userProvider.findByUsername(user.getUsername())
                             .switchIfEmpty(Maybe.error(new UserNotFoundException(user.getUsername())))
-                            .flatMapSingle(idpUser -> userProvider.updatePassword(idpUser, user.getPassword()))
+                            .flatMapSingle(idpUser -> passwordHistoryService
+                                    .addPasswordToHistory(DOMAIN, domain.getId(), user, user.getPassword() , principal, getPasswordSettings(client))
+                                    .isEmpty()
+                                    .flatMapMaybe(empty -> Maybe.just(new PasswordHistory()))
```

Comment:
I think we can replace it by `switchIfEmpty(Maybe.just(new PasswordHistory())`
wdyt?